File: android-app/app/src/main/java/com/kuklabs/kuktrip/data/trips/BudgetModels.kt
package com.kuklabs.kuktrip.data.trips

import com.google.gson.annotations.SerializedName

data class BudgetItem(
    val id: Int,
    @SerializedName("trip_id") val tripId: Int,
    val category: String = "Other",
    val name: String,
    @SerializedName("total_price") val totalPrice: Double = 0.0,
    val persons: Int? = null,
    val days: Int? = null,
    val note: String? = null,
    @SerializedName("expense_date") val expenseDate: String? = null,
    @SerializedName("reservation_id") val reservationId: Int? = null,
)

data class BudgetEnvelope(val items: List<BudgetItem> = emptyList())

data class CreateBudgetItemBody(
    val name: String,
    val category: String = "Other",
    @SerializedName("total_price") val totalPrice: Double = 0.0,
    val persons: Int? = null,
    val days: Int? = null,
    val note: String? = null,
    @SerializedName("expense_date") val expenseDate: String? = null,
)

data class BudgetItemEnvelope(val item: BudgetItem)

File: ios-app/Sources/Trips/BudgetModels.swift
import Foundation

struct TripBudgetItem: Codable, Identifiable, Hashable {
    let id: Int
    let tripId: Int
    let category: String
    let name: String
    let totalPrice: Double
    let persons: Int?
    let days: Int?
    let note: String?
    let expenseDate: String?
    let reservationId: Int?

    enum CodingKeys: String, CodingKey {
        case id, category, name, persons, days, note
        case tripId = "trip_id"
        case totalPrice = "total_price"
        case expenseDate = "expense_date"
        case reservationId = "reservation_id"
    }
}

struct BudgetEnvelope: Decodable { let items: [TripBudgetItem] }
struct BudgetItemEnvelope: Decodable { let item: TripBudgetItem }

struct CreateBudgetItemRequest: Encodable {
    let name: String
    let category: String
    let totalPrice: Double
    let persons: Int?
    let days: Int?
    let note: String?
    let expenseDate: String?

    enum CodingKeys: String, CodingKey {
        case name, category, persons, days, note
        case totalPrice = "total_price"
        case expenseDate = "expense_date"
    }
}
